lib/ruby_lsp/requests/prepare_rename.rb

```diff
@@ -12,19 +12,23 @@ class PrepareRename < Request
 
       sig do
         params(
+          global_state: GlobalState,
           document: RubyDocument,
           position: T::Hash[Symbol, T.untyped],
         ).void
       end
-      def initialize(document, position)
+      def initialize(global_state, document, position)
         super()
+        @global_state = global_state
         @document = document
         @position = T.let(position, T::Hash[Symbol, Integer])
       end
 
       sig { override.returns(T.nilable(Interface::Range)) }
       def perform
-        char_position = @document.create_scanner.find_char_position(@position)
+        char_position = @global_state.synchronize do
+          @document.create_scanner.find_char_position(@position)
```

Comment:
Should we find a way to make the mutex mandatory when calling scanner? Like:

```rb
@document.create_scanner(@global_state.mutex).find_char_position(@position)
```

And the scanner will do the mutax wrapping around `find_char_position`. This will making the requirement clear but also simplify the usage.